pkg/models/target_write_result.go

```diff
@@ -124,13 +129,26 @@ func NewTargetWriteResult(sent []*Message, failed []*Message, oversized []*Messa
 			r.MinRequestLatency = requestLatency
 		}
 		sumRequestLatency += requestLatency
+
+		var e2eLatency time.Duration
+		if !msg.CollectorTstamp.IsZero() {
```

Comment:
OK - so we just don't set the value if we don't have it set, and then we proceed - all the values will just be 0. Is that about right?

Reply:
Yeah, no collector timestamp == E2E latencies min/max/avg are 0. 

I think it's like with transformation latency? No transformations == you have zeros.

Reply:
Yeah it is the same for transformations. This wasn't a challenge to the design or anything, just validating my understanding - because I think we might want to change it in a future release (but we don't need to worry about that here).